activiti-spring-boot-starter/src/test/java/org/activiti/spring/boot/process/ProcessRuntimeBPMNTimerIT.java

```diff
@@ -0,0 +1,292 @@
+
+        ProcessInstance process = processRuntime.start(ProcessPayloadBuilder.start()
+                                                       .withProcessDefinitionKey(PROCESS_TIMER_CANCELED_EVENT)
+                                                       .build());
+        
+        String jobId = managementService.createTimerJobQuery().singleResult().getId();
+
+        assertThat(jobId).isNotNull();
+        
+        // WHEN
+        managementService.deleteTimerJob(jobId);
+
+        // THEN
+        List<BPMNTimerCanceledEvent> eventsCanceled = listenerCanceled.getEvents();
+        assertThat(eventsCanceled)
+            .extracting(BPMNTimerEvent::getEventType,
+                        BPMNTimerEvent::getProcessDefinitionId,
+                        event -> event.getEntity().getProcessDefinitionId(),
+                        event -> event.getEntity().getTimerPayload().getJobType(),
+                        event -> event.getEntity().getTimerPayload().getJobHandlerType()         
+            )
+            .contains(Tuple.tuple(BPMNTimerEvent.TimerEvents.JOB_CANCELED,
+                                  process.getProcessDefinitionId(),
+                                  process.getProcessDefinitionId(),
+                                  "timer",
+                                  "trigger-timer"
+                                  )
+             
+        );
+        
+        processRuntime.delete(ProcessPayloadBuilder.delete(process.getId()));
+    }
+    
+     
+    @Test
+    public void shouldGetTimerCanceledEventOnBoundaryEvent() throws Exception {
+        //given
+        securityUtil.logInAs("salaboy");
+        
+        ProcessInstance process = processRuntime.start(ProcessPayloadBuilder.start()
+                .withProcessDefinitionKey(PROCESS_TIMER_CANCELED_EVENT)
+                .build());
+        
+        assertThat(managementService.createTimerJobQuery().count()).isEqualTo(1);
+        
+        List<BPMNTimerScheduledEvent> eventsScheduled = listenerScheduled.getEvents();
+        assertThat(eventsScheduled)
+            .extracting(BPMNTimerEvent::getEventType,
+                        BPMNTimerEvent::getProcessDefinitionId,
+                        event -> event.getEntity().getProcessDefinitionId(),
+                        event -> event.getEntity().getTimerPayload().getJobType(),
+                        event -> event.getEntity().getTimerPayload().getJobHandlerType()         
+            )
+            .contains(Tuple.tuple(BPMNTimerEvent.TimerEvents.TIMER_SCHEDULED,
+                                  process.getProcessDefinitionId(),
+                                  process.getProcessDefinitionId(),
+                                  "timer",
+                                  "trigger-timer"
+                                  )
+             
+        );
+      
+        clear();
+        
+        
+        Page<Task> tasks = taskRuntime.tasks(Pageable.of(0, 10),TaskPayloadBuilder.tasks().build());
```

Comment:
Please, add a filter to get the tasks of the current process instance only. Otherwise, your test will fail if other tests didn't clean everything.

Reply:
Done